### PR TITLE
Raise UnlicensedFeatureError when we receive a HTTP 402

### PR DESCRIPTION
**Pull request by bdunne**
Rescue UnlicensedFeatureError when trying to fetch survey_data
https://bugzilla.redhat.com/show_bug.cgi?id=1370251

### CODE REVIEW

**Comment by bdunne** [comment]
@syncrou Please review

**Comment by syncrou** [comment]
@bdunne - This going to be the fix for `/activity_stream` as well - Or are you thinking to go another route in that case?

File: spec/support/shared_examples/job_template.rb [hunk 1]
```diff
@@ -25,6 +25,13 @@
       expect(described_class.new(api, raw_instance_no_survey).survey_spec).to be_nil
     end
   end
+
+  describe "unlicensed" do
```
Comment (syncrou) [inline spec/support/shared_examples/job_template.rb]: @bdunne - This looks great.  It would be great to have some testing around the middleware as well.

**Comment by bdunne** [comment]
@syncrou Yes, it would raise `AnsibleTowerClient::UnlicensedFeatureError` on `/activity_stream` as well.
cc @durandom @blomquisg

**Comment by bdunne** [comment]
@syncrou Middleware test added

**Comment by syncrou** [comment]
👍 LGTM

**Pull request merged**